packages/graphql/tests/integration/utils/tests-helper.ts

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Cypher from "@neo4j/cypher-builder";
+import type { ExecutionResult, GraphQLArgs } from "graphql";
+import { graphql as graphqlRuntime } from "graphql";
+import * as neo4j from "neo4j-driver";
+import { Memoize } from "typescript-memoize";
+import type { Neo4jGraphQLConstructor, Neo4jGraphQLContext } from "../../../src";
+import { Neo4jGraphQL } from "../../../src";
+import { createBearerToken } from "../../utils/create-bearer-token";
+import { UniqueType } from "../../utils/graphql-types";
+
+const INT_TEST_DB_NAME = "neo4jgraphqlinttestdatabase";
+const DEFAULT_DB = "neo4j";
+
+let helperLock = false;
+
+export class TestHelper {
+    private database: string = DEFAULT_DB;
+    private neo4jGraphQL: Neo4jGraphQL | undefined;
+    private uniqueTypes: UniqueType[] = [];
+
+    public createBearerToken(secret: string, extraData?: Record<string, any>) {
+        return createBearerToken(secret, extraData);
+    }
+
+    public async initNeo4jGraphQL(options: Omit<Neo4jGraphQLConstructor, "driver">): Promise<Neo4jGraphQL> {
+        if (helperLock) {
+            throw new Error("TestHelper still open. Did you forget calling .close()?");
+        }
+        helperLock = true;
```

Comment:
tbh, if the driver throws the tests are going to fail anyway

In any case, the lock must be before the driver (or we would have a race condition as the driver takes time to generate), I can add a try ... catch to release the lock  in case of an error, but then the close will fail so it is not as simple ...

Reply:
I see what you mean, my point was mainly due to the fact that if a driver fails on a test, then all the other tests will fail because the lock is never being released. Not important anyway!